src/firefly/js/ui/UploadTableChooser.js

```diff
@@ -82,23 +89,96 @@ function uploadSubmit(request, setUploadInfo)  {
     return false;
 }
 
-
 function existingTableSubmit(request,setUploadInfo) {
-    // todo: call this function for preloaded tables
-    const uploadInfo = {
-        serverFile: '${upload}//stuff/somename.tbl',
-        title: 'tbl title',
-        tbl_id: 'someTblId',
-        columns: [],
-        totalRows: 123,
-        tableSource: EXISTING_TBL_SOURCE,
-    };
-    setUploadInfo(uploadInfo);
-    dispatchHideDialog(dialogId);
+    if (!request) return false;
+    const tbl = getTblById('existing-table-list-ui');
+    const idx = tbl.highlightedRow;
+    const activeTblId = tbl.tableData.data[idx][3]; //tbl_id
+    const activeTbl = getTableUiByTblId(activeTblId);
+    const tableRequest = activeTbl.request;
+    const columnData = activeTbl.columns;
+    const columns = columnData.map((col) => col.visibility === 'show'? ({...col, use:true}) :  ({...col, use:false}));
+
+    const params = omitBy({
+        [ServerParams.COMMAND]: ServerParams.TABLE_SAVE,
+        [ServerParams.REQUEST]: JSON.stringify(tableRequest),
+        file_name: tableRequest.META_INFO.title,
+        file_format: Format.IPACTABLE,
+        save_to_temp: 'true'
+    }, isNil);
+
+    doJsonRequest(ServerParams.TABLE_SAVE, params).then((result) => {
+        if (!result.success) {
+            showInfoPopup('Error loading this table', result.error);
+            return false;
+        }
+        const uploadInfo = {
+            serverFile: result?.serverFile ?? null,
+            title: activeTbl.title,
+            fileName: activeTbl.title,
+            tbl_id: activeTblId,
+            columns: columns,
+            totalRows: activeTbl.totalRows,
+            tableSource: EXISTING_TBL_SOURCE,
+        };
+        setUploadInfo(uploadInfo);
+        dispatchHideDialog(dialogId);
+    });
     return false;
 }
 
+const NoTables = () => {
+    return (
```

Comment:
make the message centered, use `font-size: larger` font, and make some sort of margin-top like 40 pixels